torch/csrc/Dtype.cpp

```diff
@@ -126,6 +126,22 @@ PyTypeObject THPDtypeType = {
 
 void THPDtype_init(PyObject *module)
 {
+  // Set a __dict__ with `__module__` = `torch`. This means
+  // `__module__` value will be inherited by instances, which will
```

Comment:
This comment may need to be clarified. Like if we consider a Python program

```
class foo(object):

  def __init__(self):
    # self.__dict__ = {"__module__": "silly"}
    self.__module__ = "silly"

class bar(foo):

  def __init__(self):
    super().__init__()
```

then

```
b = bar()
b.__module__
: silly
```

so why not just set the dunder module attribute directly?

Reply:
For a C extension type, we can't set it directly in Python:
```
>>> torch.dtype.__module__ = 'torch'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can't set attributes of built-in/extension type 'torch.dtype'
>>> torch.dtype.__dict__['__module__'] = 'torch'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'mappingproxy' object does not support item assignment
```

and doing a `PyObject_SetAttrString` I believe will fail for similar reasons.

Reply:
I see; so this is the recommended way of setting these attributes on C extension types?

Sorry I didn't respond earlier, I'm on vacation at the moment but am now checking notifications.

Reply:
I couldn't find a standard idiom here. Populating `tp_dict` is explicitly supported in the docs, and user-defined types also have a `__module__` in `__dict__` so I feel comfortable that all the components of this solution are standard in isolation, but I can't rule out that there could be another approach here.